packages/core/src/components/select/select.spec.ts

```diff
@@ -331,4 +336,17 @@ describe('AtomSelect', () => {
 
     expect(handleDismiss).not.toHaveBeenCalled()
   })
+
+  it('should filter options with tag', async () => {
+    const page = await newSpecPage({
+      components: [AtomSelect],
+      html: '<atom-select />',
+    })
+
+    await page.waitForChanges()
+    const mockFiltered = optionsMock.filter((option) => option?.tag?.label)
+    const instanceObjetct = page.rootInstance.filterOptionsWithTag(optionsMock)
+
+    expect(Object.keys(instanceObjetct).length).toEqual(mockFiltered.length)
+  })
```

Comment:
It's also important to test the `setTag` method, even now you told me it is a Stencil issue.

Reply:
@felipefialho @Natalie9 I was trying to create tests to help with it, but `newSpecPage` renders a mock component version. So it renders `ion-select-option` on `page.root`, but not the `ion-item` itself (which happens on the screen). :/ 

I saw `@stencil/core/testing` has an `newE2EPage` function, but I guess it is outside of the task context. 

Alternatively, for now, guess we could mock `querySelectorAll` to return a node list and validate if those are as expected after calling `setTagInSelectOptions`. It will only help with changes on the `setTagInSelectOptions` itself

Reply:
https://github.com/juntossomosmais/atomium/pull/549/commits/1a0714d1eface0fb973f9ebdb87b6a0d3de84d40